pkg/algorithms/cmr/cmr.py

```diff
@@ -0,0 +1,123 @@
+        """Analyze the dataframe with meaning any previous data and generate percent change with a current uuid
+
+        Returns:
+            series: data series that contains attributes and full dataframe
+            change_points_by_metric: list of ChangePoints
+        """
+        logger_instance = SingletonLogger.getLogger("Orion")
+        logger_instance.info("Starting analysis using Isolation Forest")
+        self.dataframe["timestamp"] = pd.to_datetime(self.dataframe["timestamp"])
+        self.dataframe["timestamp"] = self.dataframe["timestamp"].astype(int) // 10**9
+
+        logger_instance.info('data frame ' + str(self.dataframe))
+
+        # if larger than 2 rows, need to get the mean of 0 through -2
+        self.dataframe = self.combine_data_frames( self.dataframe)
+
+        series= self.setup_series()
+        tolerancy = 20
+        metric_columns = self.metrics_config.keys()
+        df, change_points_by_metric = self.run_cmr(tolerancy, metric_columns, self.dataframe)
+        series.data= df
+        return series, change_points_by_metric
+
+
+    def run_cmr(self, tolerancy: int,metric_columns: List[str],  dataframe_list: pd.DataFrame):
+        """
+        Generate the percent difference in a 2 row dataframe
+
+        Args:
+            tolerancy (int): tolerancy to compare on 
+            metric_columns (List[str]): string list of metric column names
+            dataframe_list (pd.DataFrame): data frame of all data to compare on
+
+        Returns:
+            pd.Dataframe, dict[metric_name, ChangePoint]: Returned data frame and change points
+        """
+        change_points_by_metric={ k:[] for k in metric_columns }
+        max_date_time = pd.Timestamp.max.to_pydatetime()
+        max_time = max_date_time.timestamp()
+        difference = ["difference", max_time]
+        pass_fail_list = ["Pass/Fail", max_time]
+        for column in metric_columns:
+            pct_change_result = dataframe_list[column].pct_change()
+            single_pct_diff = round(pct_change_result.iloc[[-1]].values[0] * 100)
+            pass_fail = "Pass"
+            if single_pct_diff > tolerancy:
+                pass_fail = "Fail"
+
+            change_point = ChangePoint(metric=column,
+                                            index=1,
+                                            time=difference[1],
+                                            stats=ComparativeStats(
+                                                mean_1=dataframe_list[column][0],
+                                                mean_2=dataframe_list[column][1],
+                                                std_1=0,
+                                                std_2=0,
+                                                pvalue=1
+                                            ))
+            change_points_by_metric[column].append(change_point)
+            difference.append(single_pct_diff)
+            pass_fail_list.append(pass_fail)
```

Comment:
In the above two structures, we seem to be capturing pct_diffs & pass_fails. Why do we have `max_time` in those arrays initially?